docs(components): [upload] use new display tag (#12701)
element-plus/element-plus: docs/en-US/component/upload.md

:::

## API

### Attributes

| Name                          | Description                                                                                                                                                                           | Type                                                                                                     | Default                                                                                                            |
| ----------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | -------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------ |

> | action                        | request URL.                                                                                                                                                                          | ^[string]                                                                                                | #                                                                                                                  |

Reviewer: ```suggestion
| action ^(required)                       | request URL.                                                                                                                                                                          | ^[string]                                                                                                | #                                                                                                                  |
```

| method                        | set upload request method.                                                                                                                                                            | ^[string]                                                                                                | post                                                                                                               |
| multiple                      | whether uploading multiple files is permitted.                                                                                                                                        | ^[boolean]                                                                                               | false                                                                                                              |

> | data                          | additions options of request.                                                                                                                                                         | ^[object]`Record<string, any>`                                                                           | () => Mutable<{}>                                                                                                  |

Reviewer: The `data` attribute has been changed compared to before

> | data                          | additions options of request.                                                                                                                                                         | ^[object]`Record<string, any> \| Awaitable<Record<string, any>> \| ((rawFile: UploadRawFile) => Awaitable<Record<string, any>>)` | () => Mutable<{}>                                                                                                  |

Reviewer: ```suggestion
| data                          | additions options of request. support `Awaitable` data and `Function` since v2.3.13 | ^[object]`Record<string, any> \| Awaitable<Record<string, any>>` / ^[Function]`((rawFile: UploadRawFile) => Awaitable<Record<string, any>>)` | {}                                                                                                  |
```

Some details are missing here

Author: updated.

| drag                          | whether to activate drag and drop mode.                                                                                                                                               | ^[boolean]                                                                                               | false                                                                                                              |
| accept                        | accepted [file types](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#attr-accept), will not work when `thumbnail-mode === true`.                                     | ^[string]                                                                                                | ''                                                                                                                 |

> | on-preview                    | hook function when clicking the uploaded files.                                                                                                                                       | ^[Function]`(uploadFile: UploadFile) => void`                                                            | () => void                                                                                                         |

Reviewer: `() => void` -> `—`

What do you think?

Author: agree

| on-change                     | hook function when select file or upload file success or upload file fail.                                                                                                            | ^[Function]`(uploadFile: UploadFile, uploadFiles: UploadFiles) => void`                                  | () => void                                                                                                         |
| on-exceed                     | hook function when limit is exceeded.                                                                                                                                                 | ^[Function]`(files: File[], uploadFiles: UploadUserFile[]) => void`                                      | () => void                                                                                                         |

> | before-upload                 | hook function before uploading with the file to be uploaded as its parameter. If `false` is returned or a `Promise` is returned and then is rejected, uploading will be aborted.      | ^[Function]`(rawFile: UploadRawFile) => Awaitable<void \| undefined \| null \| boolean \| File \| Blob>` | () => void                                                                                                         |

Reviewer 2: Perhaps it can be added that it will only be triggered when `auto-upload` is set to `true`. See [comments](https://github.com/element-plus/element-plus/issues/13857#issuecomment-1663992965).

Author: updated.

| before-remove                 | hook function before removing a file with the file and file list as its parameters. If `false` is returned or a `Promise` is returned and then is rejected, removing will be aborted. | ^[Function]`(uploadFile: UploadFile, uploadFiles: UploadFiles) => Awaitable<boolean>`                    | —                                                                                                                  |

> | file-list / v-model:file-list | default uploaded files.                                                                                                                                                               | ^[object]`UploadUserFile[]`                                                                              | () => []                                                                                                           |

Reviewer: ```suggestion
| file-list / v-model:file-list | default uploaded files.                                                                                                                                                               | ^[object]`UploadUserFile[]`                                                                              | []                                                                                                           |
```

| disabled                      | whether to disable upload.                                                                                                                                                            | ^[boolean]                                                                                               | false                                                                                                              |
| limit                         | maximum number of uploads allowed.                                                                                                                                                    | ^[number]                                                                                                | —                                                                                                                  |

### Slots

> | Name    | Description                         | Scope                           |

Reviewer: ```suggestion
| Name    | Description                         | Type                           |
```
